readme.md

```diff
@@ -188,7 +188,13 @@ Type: `number` `Object`
 
 Milliseconds to wait for the server to end the response before aborting request with `ETIMEDOUT` error (a.k.a. `request` property). By default there's no timeout.
 
-This also accepts an object with separate `connect`, `socket`, and `request` fields for connection, socket, and entire request timeouts.
+This also accepts an object with separate `lookup`, `connect`, `socket`, `response` and `request` fields to specify granular timeouts for each phase of the request.
+
+- `lookup` starts when a socket is assigned and ends when the hostname has been resolved. Does not apply when using a unix domain socket.
+- `connect` starts when `lookup` completes (or when the socket is assigned if lookup does not apply to the request) and ends when the socket is connected.
+- `socket` starts when the socket is connected. See [request.setTimeout](https://nodejs.org/api/http.html#http_request_settimeout_timeout_callback).
+- `response` starts when the socket is connected and ends when the response headers are received.
+- `request` starts when the request is initiated and ends when the response's end event fires.
```

Comment:
I think the names should be as transparent as possible. lookup, connect and response are nice because they map directly to the event that ends the phase.

request and socket are the outliers, but I left them as they were to maximize backwards compatibility.

If we want to forego backwards compatibility, I'd suggest renaming `request` to `overall` or `total`. `socket` may be more expressive as `idleSocket`, but I think it's too pedantic to justify the incompatibility.

Reply:
> I'd suggest renaming request to overall or total.

👍 

> socket may be more expressive as idleSocket, but I think it's too pedantic to justify the incompatibility.

Agreed. `socket` is enough IMO.

Reply:
@sindresorhus what do you think about changing `request` to `overall` or `total`?

Reply:
Following up on this, I realized that the response timer would start too soon if the request spent significant time writing to the socket, so I implemented a fix for that by emitting a custom event on the request when the body has been written.

This also enabled the addition of a `send` timer.

If we were starting from scratch I would name the `send` timer `request` and I would name the `request` timer `overall`. But, I worry that changing the meaning of those names is subtle enough to go overlooked by users, leading to bugs that can be avoided by just keeping the imperfect names.